subsys/bluetooth/host/gatt.c

```diff
@@ -5467,26 +5467,20 @@ int bt_gatt_subscribe(struct bt_conn *conn,
 		}
 
 		/* Check if another subscription exists */
-		if (tmp->value_handle == params->value_handle &&
-		    tmp->value >= params->value) {
-			has_subscription = true;
+		if (tmp->value_handle == params->value_handle) {
+			params->value |= tmp->value;
```

Comment:
We should actually probably add a check for `params->value` to be between 0x01 and 0x03. 

E.g. 
App A calls `bt_gatt_subscribe` with value = 0x04 (not a valid value). We do not reject this, and add App A's param to the `sub->list`

While that request is outstanding, App B calls `bt_gatt_subscribe` with 0x01 (notify), but with this change, the value sent to the server become 0x05 which is also invalid. 

App A would receive a write error from the server because it sends a wrong value, but so will App B, even though it provided a valid value. 

Checking that `params->value` is a valid value before sending the request and before doing `sys_slist_prepend`